minidump/src/minidump.rs

```diff
@@ -907,9 +907,28 @@ impl Module for MinidumpModule {
     fn code_file(&self) -> Cow<'_, str> {
         Cow::Borrowed(&self.name)
     }
+
     fn code_identifier(&self) -> Cow<'_, str> {
         match self.codeview_info {
             Some(CodeView::Elf(ref raw)) => Cow::Owned(bytes_to_hex(&raw.build_id)),
+            // MacOS/Solaris minidumps. It's assumed that the age is always >0
+            // for Windows minidumps; They always start with an age of 1.
+            // Breakpad sets it to 0 on MacOS/Solaris, making it a (slightly
+            // unreliable) sentinel for those operating systems.
+            Some(CodeView::Pdb70(ref raw)) if raw.age == 0 => {
+                // Breakpad converts LC_UUIDs into GUIDs for macOS and Solaris.
+                // We convert these back to UUIDs by flipping the first 64 bytes
+                // back to big endian.
+                // https://github.com/google/breakpad/blob/main/src/client/mac/handler/minidump_generator.cc#L1468-L1486
+                // https://github.com/google/breakpad/blob/main/src/client/solaris/handler/minidump_generator.cc#L474-L488
+                let pseudo_uuid = md::GUID {
+                    data1: raw.signature.data1.swap_bytes(),
+                    data2: raw.signature.data2.swap_bytes(),
+                    data3: raw.signature.data3.swap_bytes(),
+                    data4: raw.signature.data4,
+                };
```

Comment:
We have a bunch of endianess machinery between `scroll` and the stdlib, can we not leverage that here to do this a bit less messily..?

Reply:
Were there specific helpers that you had in mind that could be used to do this a little more elegantly? 

There's also a bit of ambiguity here that I'd like to clarify: Are you asking that the
1. Entire GUID is re-read with the correct endianness
2. `data1-3` are swapped using `scroll` instead of `swap_bytes`?

Reply:
fwiw it looks like https://github.com/luser/rust-minidump/pull/438 also adds the uuid crate as a dependency, which opens up the option of using [`Uuid::from_fields_le`](https://docs.rs/uuid/0.8.2/uuid/struct.Uuid.html#method.from_fields_le) that appears to be designed for the exact conversion that's being performed in this block. Maybe that's the best option to go with here?

Reply:
That sounds good, let's wait until that patch lands

Reply:
So I did some testing with symbolicator and it looks like we didn't need this swapping after all. This gets rid of the ugly bits pointed out in the original comment and removes the loose dependency this PR has on #438. 🎉